keycloak-event-emitter-tests/src/test/java/io/cloudtrust/keycloak/kafkaeventemitter/KafkaEventEmitterProviderTest.java

```diff
@@ -0,0 +1,134 @@
+        userEntity.setUsername("test-user");
+        UserModel user = new UserAdapter(null, null, null, userEntity);
+        Mockito.when(keycloakSession.users().getUserById((RealmModel) Mockito.any(), Mockito.any())).thenReturn(user);
+
+        // Realm
+        RealmEntity realmEntity = new RealmEntity();
+        realmEntity.setId("realmId");
+        RealmModel realm = new RealmAdapter(null, null, realmEntity);
+        Mockito.when(keycloakSession.realms().getRealm(Mockito.any())).thenReturn(realm);
+
+        mockProducer = new MockProducer<>(true, new StringSerializer(), new StringSerializer());
+    }
+
+    @Test
+    public void testEventFlatbufferFormatOutput() {
+        IdGenerator idGenerator = new IdGenerator(1, 1);
+        LinkedBlockingQueue<ProducerRecord<String, String>> pendingEvents = new LinkedBlockingQueue<>(50);
+
+        KafkaEventEmitterProvider kafkaEventEmitterProvider = new KafkaEventEmitterProvider(keycloakSession, mockProducer, topicEvent, topicAdminEvent, idGenerator, pendingEvents, KafkaEventEmitterState.WORKING);
+
+        Event event = createEvent();
+        kafkaEventEmitterProvider.onEvent(event);
+        List<ProducerRecord<String, String>> recordList = mockProducer.history();
+
+        ProducerRecord<String, String> producedEvent = recordList.get(0);
+
+        byte[] b = Base64.getDecoder().decode(producedEvent.value());
+        flatbuffers.events.Event receivedEvent = flatbuffers.events.Event.getRootAsEvent(ByteBuffer.wrap(b));
+        Assert.assertEquals(event.getTime(), receivedEvent.time());
+        Assert.assertEquals(event.getType().ordinal(), receivedEvent.type());
+        Assert.assertEquals(event.getClientId(), receivedEvent.clientId());
+    }
+
+    @Test
+    public void testAdminEventFlatbufferFormatOutput() {
+        IdGenerator idGenerator = new IdGenerator(1, 1);
+        LinkedBlockingQueue<ProducerRecord<String, String>> pendingEvents = new LinkedBlockingQueue<>(50);
+
+        KafkaEventEmitterProvider kafkaEventEmitterProvider = new KafkaEventEmitterProvider(keycloakSession, mockProducer, topicEvent, topicAdminEvent, idGenerator, pendingEvents, KafkaEventEmitterState.WORKING);
+
+
+        AdminEvent event = createAdminEvent();
+        kafkaEventEmitterProvider.onEvent(event, false);
+        List<ProducerRecord<String, String>> recordList = mockProducer.history();
+
+        ProducerRecord<String, String> producedEvent = recordList.get(0);
+
+        byte[] b = Base64.getDecoder().decode(producedEvent.value());
+
+        flatbuffers.events.AdminEvent receivedEvent = flatbuffers.events.AdminEvent.getRootAsAdminEvent(ByteBuffer.wrap(b));
+        Assert.assertEquals(event.getTime(), receivedEvent.time());
+        Assert.assertEquals(event.getOperationType().ordinal(), receivedEvent.operationType());
+        Assert.assertEquals(event.getAuthDetails().getUserId(), receivedEvent.authDetails().userId());
+    }
+
+    @Test
+    public void testNoConnection() {
```

Comment:
It might be intresting to have a test for the factory if possible and test what happen if no connection at this time ?